services/userService.js

```diff
@@ -0,0 +1,61 @@
+const User = require("../models/user");
+
+async function createUser(data) {
+  try {
+    const user = await User.create(data);
+    return user;
+  } catch (error) {
+    throw error;
+  }
+}
+
+async function getUserById(id) {
+  try {
+    const user = await User.findByPk(id);
+    if (!user) {
+      throw new Error("User not found");
+    }
+    return user;
+  } catch (error) {
+    throw error;
+  }
+}
+
+async function getAllUsers() {
+  try {
+    const users = await User.findAll();
+    return users;
+  } catch (error) {
+    throw error;
+  }
+}
+
+async function updateUser(id, data) {
+  try {
+    const user = await User.findByPk(id);
+    if (!user) {
+      throw new Error("User not found");
+    }
+    await user.update(data);
+    return user;
+  } catch (error) {
```

Comment:
Error handling should be more informative. Instead of just throwing the error, consider logging the error or providing more context about what went wrong.